examples/remote/binary_dependencies/cargo/crates.bzl

```diff
@@ -9,6 +9,144 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # bui
+    },
+}
+
+def crates(deps):
+    """EXPERIMENTAL -- MAY CHANGE AT ANY TIME: Finds the fully qualified label of the requested crates for the package where this macro is called.
+
+    WARNING: This macro is part of an expeirmental API and is subject to change.
+
+    Args:
+        deps (list or str): Either a list of dependencies or a string of one which will
+            be converted into a list.
+    Returns:
+        list: A list of labels to cargo-raze generated targets (str)
+    """
+
+    # Join both sets of dependencies
+    dependencies = dict()
+    for dep_map in [_DEPENDENCIES, _PROC_MACRO_DEPENDENCIES, _DEV_DEPENDENCIES, _DEV_PROC_MACRO_DEPENDENCIES]:
+        for package_name in _DEPENDENCIES:
+            if package_name in dependencies:
+                dependencies[package_name].extend(dep_map[package_name])
+            else:
+                dependencies[package_name].update(dep_map[package_name])
+
+    if not deps:
+        fail("An invalid argument has been provided. Please pass a crate name or a list of crate names")
+
+    if not dependencies:
+        return []
+
+    if type(deps) == "string":
+        deps = [deps]
+
+    errors = []
+    crates = []
+    for crate in deps:
+        if crate not in dependencies[native.package_name()]:
+            errors.append(crate)
+        else:
+            crates.append(dependencies[native.package_name()][crate])
+
+    if errors:
+        fail("Missing crates `{}` for package `{}`. Available crates `{}".format(
+            errors,
+            native.package_name(),
+            dependencies[native.package_name()],
+        ))
+
+    return crates
+
+def all_crates(normal = False, proc_macro = False, dev = False, dev_only = False):
+    """EXPERIMENTAL -- MAY CHANGE AT ANY TIME: Finds the fully qualified label of all requested direct crate dependencies \
+    for the package where this macro is called.
+
+    If no parameters are set, all normal and proc_macro dependencies are returned.
+    Setting any one flag will otherwise impact the contents of the returned list
+
+    WARNING: This macro is part of an expeirmental API and is subject to change.
+
+    Args:
+        normal (bool, optional): If True, normal dependencies are included in the
+            output list. Defaults to False.
+        proc_macro (bool, optional): If True, proc_macro dependencies will be
+            included in the output list. Defaults to False.
+        dev (bool, optional): If True, dev dependencies are included when the
+            `normal` and `proc_macro` parameters are used. Defaults to False.
+        dev_only (bool, optional): If True, only development dependencies will be
+            returned by this list. This paramter otherwise follows the same rules
+            as `dev`. Defaults to False.
+
+    Returns:
+        list: A list of labels to cargo-raze generated targets (str)
+    """
+
+    # Determine the relevant maps to use
+    dependencies = dict()
+    all_maps = []
```

Comment:
I think this needs a better name. 

all_deps or all_dependencies?

Reply:
This represents a grouping of all dependency maps, which are then iterated over. So it's not exactly `all_deps` but it does technically contain that data. Still change?

Reply:
yes, or alternative, a more specific and accurate name if you have one. "all_maps" isn't a good name.

Reply:
`all_dependency_maps`?

Reply:
sgtm